src/remoted/manager.c

```diff
@@ -582,6 +498,78 @@ void save_controlmsg(const keyentry * key, char *r_msg, size_t msg_length, int *
     os_free(clean);
 }
 
+/* Assign a group to an agent without group */
+cJSON *assign_group_to_agent(const char *agent_id, const char *md5) {
+    cJSON *result = NULL;
+    char* group = NULL;
+
+    os_calloc(OS_SIZE_65536 + 1, sizeof(char), group);
+
+    mdebug2("Agent '%s' with file '%s' MD5 '%s'", agent_id, SHAREDCFG_FILENAME, md5);
+
+    w_mutex_lock(&files_mutex);
+
+    if (!guess_agent_group || (!find_group_from_file(SHAREDCFG_FILENAME, md5, group) && !find_multi_group_from_file(SHAREDCFG_FILENAME, md5, group))) {
+        // If the group could not be guessed, set to "default"
+        // or if the user requested not to guess the group, through the internal
+        // option 'guess_agent_group', set to "default"
+        strncpy(group, "default", OS_SIZE_65536);
+    }
+
+    w_mutex_unlock(&files_mutex);
+
+    wdb_set_agent_groups_csv(atoi(agent_id),
+                            group,
+                            WDB_GROUP_MODE_EMPTY_ONLY,
+                            w_is_single_node(NULL) ? "synced" : "syncreq",
+                            NULL);
+
+    mdebug2("Group assigned: '%s'", group);
+
+    result = cJSON_CreateObject();
+    cJSON_AddStringToObject(result, "group", group);
+
+    os_free(group);
+    return result;
+}
+
+STATIC cJSON *assign_group_to_agent_worker(const char *agent_id, const char *md5) {
+    char response[OS_MAXSTR] = "";
+    char *request = NULL;
+    cJSON *payload_json = NULL;
+    cJSON *response_json = NULL;
+    cJSON *data_json = NULL;
+
+    cJSON *message_json = cJSON_CreateObject();
+    cJSON *parameters_json = cJSON_CreateObject();
+
+    cJSON_AddStringToObject(parameters_json, "agent", agent_id);
+    cJSON_AddStringToObject(parameters_json, "md5", md5);
+
+    cJSON_AddStringToObject(message_json, "command", "assigngroup");
+    cJSON_AddItemToObject(message_json, "parameters", parameters_json);
+
+    payload_json = w_create_sendsync_payload("remoted", message_json);
+
+    request = cJSON_PrintUnformatted(payload_json);
+
+    mdebug2("Sending message to master node: '%s'", request);
+
+    w_send_clustered_message("sendsync", request, response);
```

Comment:
Why don't you think it's necessary to check the return value of the `w_send_clustered_message()` function?
If there was an error, the `cJSON_Parse()` function may fail.